### DISCUSSION TITLE
How are you debugging the Keystone

### DISCUSSION THREAD

**Discussion by pahaz** [keystonejs/keystone#7262] [Questions]
So, there are often many cases where you need to set breakpoints inside the Keystone while developing your own app.

Currently, there is no source map inside the keystone-6 package, and there is no "src" inside the package.

It would be useful to add some documentation related to debugging.

**Comment by dcousens** [comment]
Hi @pahaz, 
Please refer to [how to debug nodejs](https://nodejs.org/api/debugger.html),  Keystone isn't anything particularly special in that regard.
The Typescript is compiled to Javascript,  but the compiled source code is still readable.

**Reply by pahaz** [reply]
@dcousens thanks for your answer. But, my question is not about hot-to debug the node.js application. It's about the Keystone.

If you try to debug the Keystone APP. The first thing for that is the source map or internal source code. Currently, the Keyston package doesn't have `src` or a source map.

As for me, I can just add something like:
```
`{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@keystone-6/core": ["../keystone6/packages/core/src"],
      "@keystone-6/core/*": ["../keystone6/packages/core/src/*"]
    }
  }
}
```

to My `tsconfig.json`. But is there an expected way for the Keystone debugging? I think No. 

Because I was faced with some problems related to tricky source code import logic.

**Reply by raveling** [reply]
@pahaz 

Are you trying to debug Keystone in the context of your own App? Or in the context of the Keystone repo...

**Reply by gwyneplaine** [reply]
Sorry @pahaz, are you trying to debug code in your app, or are you trying to debug keystone code used by your app? If the former, you shouldn't need keystone source or sourcemaps.

**Reply by pahaz** [reply]
To be clear, I'm trying to debug the Keystone core in a context of developing my own App.
